src/reader/parser/inside_doctype.rs

```diff
@@ -6,6 +6,10 @@ use super::{DoctypeSubstate, PullParser, QuoteToken, Result, State};
 
 impl PullParser {
     pub fn inside_doctype(&mut self, t: Token, substate: DoctypeSubstate) -> Option<Result> {
+        if let Some(ref mut doctype) = self.data.doctype {
+            doctype.push_str(t.to_string().as_str());
```

Comment:
Avoids allocation `use fmt::Write`:

```suggestion
            write!(&mut doctype, "{t}").ok()?;
```